lib/service/postcard_chat_service.dart

```diff
@@ -0,0 +1,172 @@
+import 'dart:convert';
+import 'dart:io';
+import 'dart:typed_data';
+
+import 'package:autonomy_flutter/common/environment.dart';
+import 'package:autonomy_flutter/common/injector.dart';
+import 'package:autonomy_flutter/model/chat_message.dart' as app;
+import 'package:autonomy_flutter/model/pair.dart';
+import 'package:autonomy_flutter/service/chat_auth_service.dart';
+import 'package:autonomy_flutter/service/tezos_service.dart';
+import 'package:autonomy_flutter/util/log.dart';
+import 'package:crypto/crypto.dart';
+import 'package:eth_sig_util/util/utils.dart';
+import 'package:flutter_chat_types/flutter_chat_types.dart' as types;
+import 'package:libauk_dart/libauk_dart.dart';
+import 'package:web_socket_channel/io.dart';
+import 'package:web_socket_channel/web_socket_channel.dart';
+
+class PostcardChatService {
+  final bool maintainConnection;
+
+  //constructor
+  PostcardChatService({required this.maintainConnection});
+
+  WebSocketChannel? _websocketChannel;
+  String? _address;
+  String? _id;
+  Pair<WalletStorage, int>? _wallet;
+
+  Future<void> connect({
+    required String address,
+    required String id,
+    required Pair<WalletStorage, int> wallet,
+  }) async {
+    log.info("[CHAT] connect: $address, $id");
+    if (_address == address &&
+        _id == id &&
+        _wallet == wallet &&
+        _websocketChannel != null) {
+      log.info("[CHAT] connect to the same channel. Do nothing");
+      return;
+    }
+    _address = address;
+    _id = id;
+    _wallet = wallet;
+    final link = "/v1/chat/ws?index_id=$id&address=$address";
+    final header = await _getHeader(link, wallet, address);
+    _websocketChannel = IOWebSocketChannel.connect(
+      "${Environment.postcardChatServerUrl}$link",
+      headers: header,
+      customClient: HttpClient(),
+      pingInterval: const Duration(seconds: 50),
+    );
+    await _websocketChannel?.ready;
+  }
+
+  void addListener({
```

Comment:
consider wrapping these listener functions into an object like (ChatListener).
- We should support `addListener` and `removeListener`.
- Then we can allow & maintain multiple listeners for a single `ChatService`.